### DISCUSSION TITLE
Deleting a file.

### DISCUSSION THREAD

**Discussion by anthonypaulbalducci** [community/community#78548] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi... So this should be like 'super simple' and I even read the help doc.

I need to remove a few files from my account just because I put them in the wrong directory. I mean I should be for all intensive puposes logged in to my account, and from the '...' drop-down I can see the 'delete file' link....

But it is greyed out... What gives ?

**Comment by DNin01** [comment]
It might be that you're viewing a commit or changes as you're trying to do this, and you can't delete the file from there. Instead, click the "Code" tab to go back to the root of the repository, then make sure you are on a branch (like `main`). If you're not, select it by clicking this button:

![Screenshot of the repository page. A dropdown menu, labeled with a branch icon and "main", is highlighted with an orange outline.](https://github.com/github/docs/blob/main/assets/images/help/branches/branch-selection-dropdown-global-nav-update.png?raw=true)

Now, navigate to the directory where the files were stored and you should have the option to delete them.

<br>

You can also use Git or [GitHub Desktop](https://desktop.github.com/) to revert the commit that added these files, and then place them back in the correct location.

**Comment by anthonypaulbalducci** [comment]
Okay so I checked out and tried what you said-- I am super on the 'main'
branch of the repository and I am sorry these files are labeled 'malware'.
It is obviously not malware, yet a course project written in R about
'detecting' malware. The subfiles are where they should be. That is fine.

I'm just still having trouble deleting my mistakes uploading on the main
branch.

Best,
-A

**Comment by anthonypaulbalducci** [comment]
Okay, thank you.

For some reason it was greyed out at the time but then I also realized I
had to run the commit. Fixed now.

Best,
-A
